spec/expression.dd

```diff
@@ -96,10 +96,38 @@ $(H4 Assignment Operator Expressions)
     $(P except that:)
 
     $(UL
-        $(LI operand $(D a) is only evaluated once)
+        $(LI operand $(D a) is only evaluated once, see below.)
         $(LI overloading $(I op) uses a different function than overloading $(I op)= does)
         $(LI the left operand of $(D >>>=) does not undergo integral promotions before shifting)
     )
+    
+    $(P The expression evaluates according to
+
+        --------------
+        void doOpAssignment(ref int x, int y)
+        {
+            x = x op y;
+        }
+        doOpAssignment(a, b);
```

Comment:
The description needs to include the cast. I'd say a correct formulation goes as follows:

```d
$(P The expression `a @= b`, where `@` stands for any of the binary operators, is evaluated
as if it were lowered as follows:

(function ref(ref x, y) { return x = cast(typeof(x)) (x @ y); })(a, b)
```

See also (sadly) https://issues.dlang.org/show_bug.cgi?id=16271 which blocks this. Other than that this definition is really solid because defines everything by "redirecting" to function calls.